### PR TITLE
Issue #52938 fix: blind people can tell how full they feel when eating

### LINKED ISSUE #52938: blind people can't tell how full they feel when eating
Issue reported from  Round ID: [144262](https://scrubby.melonmesa.com/round/144262) (/tg/Station Manuel [ENGLISH] [US-WEST] [ROLEPLAY] [99% LAG FREE])

Reporting client version: 513.1527

this makes it a guessing game to tell if you are full yet as you can't stop when unwillingly eating

### PR DESCRIPTION
**Pull request by BarrelOfNapkins**
First PR in a while, if there' something I'm missing just let me know.

## About The Pull Request

Fixes #52938

Now if you're blind you can tell how full you're getting.
Reformatted the else/if logic to remove some repetitive checks, and added some vars on the stack to make things a bit more transparent.

## Why It's Good For The Game

Blind characters should feel how full they feel while eating.

## Changelog
:cl:
fix: if a character is blind, they can feel how full they feel while eating
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
This will display the blind message to all blind people in the visible area, not just the person eating the food.

**Comment by BarrelOfNapkins** [comment]
> This will display the blind message to all blind people in the visible area, not just the person eating the food.
> 
> (the third argument of `visible_message`, `blind_message`, is the message displayed to all people who would normally receive the message but are blind, including themselves if they're the person who the message is being sent from.)

Ah thanks ... I don't really have any way to test for this on my own; pushed an update

File: code/datums/components/food/edible.dm [hunk 1]
```diff
@@ -275,20 +275,40 @@ Behavior that's still missing from this component that original food items had t
...
+			message_to_nearby_audience = span_notice("[eater] hungrily [eatverb]s \the [parent].")
+			message_to_consumer = span_notice("You hungrily [eatverb] \the [parent].")
+		else
+			message_to_nearby_audience = span_notice("[eater] hungrily [eatverb]s \the [parent], gobbling it down!")
+			message_to_consumer = span_notice("You hungrily [eatverb] \the [parent], gobbling it down!")
+
+		//if we're blind, we want to feel how hungrily we ate that food
+		message_to_blind_consumer = message_to_consumer
+		eater.show_message(message_to_consumer, MSG_VISUAL, message_to_blind_consumer)
+		eater.visible_message(message_to_nearby_audience)
```
Comment (MrMelbert) [inline code/datums/components/food/edible.dm:310]:
```suggestion
		eater.visible_message(message_to_nearby_audience, ignored_mobs = eater)
```
The eater can see their own visible message if you don't ignore them, so non-blind people would see both the `nearbly_audience` message and the `consumer` message

Comment (BarrelOfNapkins) [reply code/datums/components/food/edible.dm:310]: Ah wow you're right, I had not expected visible_message to alert a src; updated

Comment (MrMelbert) [reply code/datums/components/food/edible.dm]: Welcome to my frustrations (#51941)

**Pull request merged**